src/app/dev/DevToys.Tools/Tools/Text/ListCompare/ListCompareGuiTool.cs

```diff
@@ -0,0 +1,215 @@
+                                            .Title(ListCompare.ListB)
+                                            .OnTextChanged(OnListBChanged)))
+                            .WithBottomPaneChild(
+                                SplitGrid()
+                                    .Vertical()
+
+                                    .WithLeftPaneChild(
+                                        _diffListInputAInterB
+                                            .Title(ListCompare.AInterB)
+                                            .CompareModeAInterB()
+                                            .ReadOnly()
+                                            .Extendable())
+                                    .WithRightPaneChild(
+                                        SplitGrid()
+                                        .Vertical()
+                                         .WithLeftPaneChild(
+                                            _diffListInputAOnly
+                                                .Title(ListCompare.AOnly)
+                                                .CompareModeAOnly()
+                                                .ReadOnly()
+                                                .Extendable())
+
+                                        .WithRightPaneChild(
+                                            _diffListInputBOnly
+                                            .Title(ListCompare.BOnly)
+                                            .CompareModeBOnly()
+                                            .ReadOnly()
+                                            .Extendable())
+                                    )
+                            ))));
+
+    public void OnDataReceived(string dataTypeName, object? parsedData)
+    {
+    }
+
+    private ValueTask OnCaseSensitiveChanged(bool caseSensitive)
+    {
+        StartCompare();
+        return ValueTask.CompletedTask;
+    }
+
+    private ValueTask OnListAChanged(string listA)
+    {
+        _listA = listA;
+        StartCompare();
+        return ValueTask.CompletedTask;
+    }
+
+    private ValueTask OnListBChanged(string listB)
+    {
+        _listB = listB;
+        StartCompare();
+        return ValueTask.CompletedTask;
+    }
+
+
+    private void StartCompare()
+    {
+        _cancellationTokenSource?.Cancel();
+        _cancellationTokenSource?.Dispose();
+        _cancellationTokenSource = new CancellationTokenSource();
+
+        WorkTask = CompareAsync(_listA, _listB, _settingsProvider.GetSetting(caseSensitive), _cancellationTokenSource.Token);
+    }
+
+    private async Task CompareAsync(string listA, string listB, bool caseSensitive, CancellationToken cancellationToken)
+    {
+        using (await _semaphore.WaitAsync(cancellationToken))
+        {
+            await TaskSchedulerAwaiter.SwitchOffMainThreadAsync(cancellationToken);
+
+            ResultInfo<string> compareResultAInterB = await ListCompareHelper.CompareAsync(
```

Comment:
To improve speed, instead of chaining calls, we could run them in parallel and wait that they all finish at once. Like here:
https://github.com/veler/DevToys/blob/6fa986b48351eb4f9a86bacc9617652f465ff8ee/src/app/dev/DevToys.Tools/Tools/Graphic/ColorBlindnessSimulator/ColorBlindnessSimulatorGuiTool.cs#L224-L258

Reply:
Ok, so i've removed the "async" side in the the helper to keep it only on GUI side, right ?

Reply:
Yes! That's perfect 😁